src/libecalc/presentation/yaml/yaml_models/pyyaml_yaml_model.py

```diff
@@ -145,14 +145,27 @@ def __init__(
                 )
 
         def load(self, yaml_file: ResourceStream):
+            class UniqueKeyLoader(self.__loader):  # type: ignore[name-defined]
+                def construct_mapping(self, node, deep=False):
+                    mapping = set()
+                    for key_node, _ in node.value:
```

Comment:
is this code exactly as SafeLoader is, except that it checks explicitly for duplicate keys at each level?

Reply:
The way I understand it self.__loader is SafeLoader with the !include constructor added. And this also adds checking for duplicate keys to self._loader.

Reply:
I see now that you call super(). I didnt see that earlier. So it calls SafeConstructor´s construct_mapping it seems, so then I think it is good